packages/interactivity/src/hooks.tsx

```diff
@@ -260,18 +261,35 @@ export const directive = (
 
 // Resolve the path to some property of the store object.
 const resolve = ( path, namespace ) => {
+	if ( namespace === '' ) {
+		if ( isDebug ) {
+			// eslint-disable-next-line no-console
+			console.warn(
+				`Namespace cannot be an empty string. Error found when trying to use "${ path }"`
+			);
+		}
+		return;
+	}
 	let resolvedStore = stores.get( namespace );
 	if ( typeof resolvedStore === 'undefined' ) {
 		resolvedStore = store( namespace, undefined, {
 			lock: universalUnlock,
 		} );
 	}
-	let current = {
+	const current = {
 		...resolvedStore,
 		context: getScope().context[ namespace ],
 	};
-	path.split( '.' ).forEach( ( p ) => ( current = current[ p ] ) );
-	return current;
+	try {
+		return path.split( '.' ).reduce( ( acc, key ) => acc[ key ], current );
+	} catch ( e ) {
+		if ( isDebug ) {
+			// eslint-disable-next-line no-console
+			console.warn(
+				`There was an error when trying to resolve the path "${ path }" in the namespace "${ namespace }".`
```

Comment:
We need to support lazy/dynamically initialized stores. This means is that instead of throwing an error, we need to subscribe to the store. This way, when the store is finally loaded, the component will re-render, and everything will work.

I think @DAreRodz was working on it. 

To make it as efficient as possible, we need to subscribe as deeply as possible. DeepSignal can subscribe to properties that have not been defined. So, for example, if you are trying to access state.obj.prop and obj is not defined, we should subscribe to state.obj. It shouldn't be complex to do. We just have to let it keep subscribing (which it does automatically) until it throws the error. So basically it's just catching the error.

Reply:
Is the `acc[ key ]` access as deep as it can go before erroring sufficient to subscribe?

Reply:
Yes. It subscribes to the `undefined` value of the `key` key. Once that key is replaced with a real value, primitive or object, the computations are invalidated and everything rerenders, resubscribing to the new values (deeper this time if key is an object).